ccmlib/containers/docker_client.py

```diff
@@ -0,0 +1,143 @@
+                if is_remove_current_container and self.container.name == container.name:
+                    continue
+                container.remove(force=True)
+
+    def create_container(self, ports=None, volumes=None, is_clear_container=True, **kwargs):
+        if is_clear_container:
+            self.remove_old_containers()
+        LOGGER.debug("Creating container using image '{}'".format(self.image))
+        self.container = self.client.containers.create(
+            image=self.image, hostname=self.name, detach=True, stdin_open=True, tty=True, ports=ports, volumes=volumes,
+            name=self.name, network_disabled=False, privileged=True, pid_mode="host", **kwargs)
+        LOGGER.debug("Start container '{}'".format(self.name))
+        self.container.start()
+        return self.container.attrs["State"]["Pid"], self.container.name
+
+    def close(self):
+        if not self.container or self.is_running():
+            raise docker.errors.DockerException("The '{}' docker was closed!".format(self.container.name))
+
+        container_name = self.container.name
+        LOGGER.debug(
+            "Stop the '{}' container (The current state is '{}') is".format(container_name, self.container.status))
+        self.container.stop(timeout=1)
+        LOGGER.debug("Remove the '{}' container".format(container_name))
+        try:
+            self.container.remove(force=True)
+        except docker.errors.APIError as err:
+            LOGGER.warning("Failed to remove '{}' container".format(self.image), exc_info=err)
+        self.container = self.container_name = self.container_pid = None
+        LOGGER.info("The'{}' container is closed".format(self.name))
+
+    def execute_command(self, command, is_stream=True, **kwargs):
+        if self.container is None:
+            raise docker.errors.DockerException("container is not running".format(self.name, command))
+
+        command = f"{self.shell_type} -c \"{command}\""
+        LOGGER.debug("Execute '{}' command in '{}' container".format(command, self.name))
+        try:
+            output = self.container.exec_run(command, stream=is_stream, **kwargs)[1]
+        except docker.errors.APIError as err:
+            if not self.is_running():
+                self.close()
+            raise docker.errors.DockerException("Docker container API error: {}".format(err))
+        if is_stream:
+            response = "".join([line.decode("utf-8", "ignore").rstrip() for line in output])
+            LOGGER.debug("The response of '{}' command is:\n{}".format(command, response))
+            output = response
+
+        return output
+
+    def is_running(self):
+        """
+        Check if class container state is running
+        :return: boolean
+        """
+        try:
+            return self.container.status == "running"
+        except docker.errors.APIError:
+            return False
+
+    @staticmethod
+    def server_ip(destination_address="8.8.8.8"):
```

Comment:
what this function is doing ? and for which propose ?

Reply:
This method returns the node public IP